raytracing/rays.py

```diff
@@ -22,6 +22,7 @@
 MatrixGroup or any tracing function. 
 """
 
+
 class Rays:
     def __init__(self, rays=None):
         if rays is None:
```

Comment:
Rays as a base class wants an array _rays, but it should not be writable.
RandomRays will create rays as it goes because if not, it is too lengthy. It will append them to the array as they are created.